src/vs/workbench/contrib/chat/browser/chatPasteProviders.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from '../../../../base/common/cancellation.js';
+import { IReadonlyVSDataTransfer } from '../../../../base/common/dataTransfer.js';
+import { HierarchicalKind } from '../../../../base/common/hierarchicalKind.js';
+import { IRange } from '../../../../editor/common/core/range.js';
+import { DocumentPasteContext, DocumentPasteEditProvider, DocumentPasteEditsSession } from '../../../../editor/common/languages.js';
+import { ITextModel } from '../../../../editor/common/model.js';
+
+
+export class PasteImageProvider implements DocumentPasteEditProvider {
```

Comment:
Should we have a `ChatPasteEditProvider` for all kinds of chat paste logic instead of having one specific to images and another for resources...? or is this the recommended way to do this?

Reply:
Matt pointed me to https://github.com/microsoft/vscode/blob/8154d87ae88246acb57adb56e18b5e579daa7628/src/vs/editor/contrib/dropOrPasteInto/browser/defaultProviders.ts#L88

I think the logic will be a little different between images and files but, alternatively if the logic is similar, you could probably bundle the mime types together like `public readonly pasteMimeTypes = ['image/*', ;fileTypehere']` and we can rename it